src/shared/spawn-polkit-agent.c

```diff
@@ -37,22 +37,30 @@ int polkit_agent_open(void) {
         if (!is_main_thread())
                 return -EPERM;
 
+        if (access(POLKIT_AGENT_BINARY_PATH, X_OK) < 0)
+                return log_error_errno(errno, "Failed to find polkit agent binary. Is polkit agent installed?");
```

Comment:
uh, it's entirely fine if pk isn't around, we shouldn't complain about things then. Of course you cannot issue priv operations as unpriv user then, but that's totally ok, and this really should not be a problem.

We fail gracefully here if PK is not around, and that's *good*, because we don't really want to force people to install pk

Reply:
I'm not sure how to fix this then. The problem is that it fails silently without even mentioning anything about polkit. Perhaps just the change to execv is enough?

Reply:
Logging when execv fails should be fine to do, as it sounds generally useful yes, so I'd say drop this part and we can merge the other